### PR TITLE
NNPDENS and NNPDEHan Integration from NeuralPDE.jl

### PR DESCRIPTION
**Pull request by ashutosh-b-b**

### CODE REVIEW

File: src/NNPDENS.jl [hunk 1]
```diff
@@ -210,6 +211,11 @@ function DiffEqBase.solve(
             end
         end
         u_low = sum(exp(I)*g(X) - Q for (I ,Q ,X) in sol_low())/(trajectories_lower)
-        save_everystep ? iters : re1(p3)(x0)[1] , u_low , u_high
+        if save_everystep
```
Comment (vboussange) [inline src/NNPDENS.jl:214]: Similar comments to NNPDEHan.jl apply here.

Comment (vboussange) [inline src/NNPDENS.jl:214]: Similar comments to NNPDEHan.jl apply here.

**Review by vboussange — CHANGES_REQUESTED** [review]
Hey, thanks a lot for the work! Looks like everything runs smoothly, but it would be great to a bit of documentation.

File: src/NNPDEHan.jl [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                    a_ = A[findmax(collect(A).*u .- collect(legendre_transform(f_matrix, a, u_domain) for a in A))[2]]
+                    I = I + a_*dt
+                    Q = Q + exp(I)*legendre_transform(f_matrix, a_, u_domain)
+                end
+                I , Q , X
+            end
+        end
+        u_low = sum(exp(I)*g(X) - Q for (I ,Q ,X) in sol_low())/(trajectories_lower)
+        if save_everystep
+            sol = PIDESolution(X0, ts, losses, iters, u0(X0)[1], (u_low, u_high))
```
[the remark this answers is not in the document]
Comment (ashutosh-b-b) [reply src/NNPDEHan.jl:150]: Hi! These basically provide upper and lower bounds to the solution, you can check more about it [here](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3071506) . I forgot to push the `HighDimPDE.jl` and added a field.

File: src/NNPDEHan.jl [hunk 3]
```diff
@@ -0,0 +1,157 @@
+struct NNPDEHan{C1,C2,O} <: HighDimPDEAlgorithm
+    u0::C1
+    σᵀ∇u::C2
+    opt::O
+end
+NNPDEHan(u0,σᵀ∇u;opt=Flux.ADAM(0.1)) = NNPDEHan(u0,σᵀ∇u,opt)
```
[the remark this answers is not in the document]
Comment (ashutosh-b-b) [reply src/NNPDEHan.jl:6]: [Here](https://www.pnas.org/doi/10.1073/pnas.1718942115)

**Review by vboussange — COMMENTED** [review]
Thanks for the clarification. I will add some documentation and push it to the main branch. Thanks again!

**Pull request merged**